program/src/accept_offer/mod.rs

```diff
@@ -0,0 +1,415 @@
+        ctx.accounts.auction_house_treasury.to_account_metas(None),
+        ctx.accounts.buyer_trade_state.to_account_metas(None),
+        ctx.accounts.seller_trade_state.to_account_metas(None),
+        ctx.accounts.free_seller_trade_state.to_account_metas(None),
+        ctx.accounts.reward_center.to_account_metas(None),
+        ctx.accounts
+            .reward_center_reward_token_account
+            .to_account_metas(None),
+        ctx.accounts.ah_auctioneer_pda.to_account_metas(None),
+    ]
+    .iter()
+    .flat_map(|v| v.iter().cloned())
+    .collect::<Vec<AccountMeta>>()
+}
+
+pub fn handler(
+    ctx: Context<AcceptOffer>,
+    execute_sale_params: ExecuteSaleParams,
+    auctioneer_sell_params: AuctioneerSellParams,
+) -> Result<()> {
+    let auction_house_key = ctx.accounts.auction_house.key();
+    let reward_center_bump = ctx.accounts.reward_center.bump;
+    let auctioneer_sell_ctx_accounts = AuctioneerSell {
+        metadata: ctx.accounts.metadata.to_account_info(),
+        wallet: ctx.accounts.seller.to_account_info(),
+        token_account: ctx.accounts.token_account.to_account_info(),
+        auction_house: ctx.accounts.auction_house.to_account_info(),
+        auction_house_fee_account: ctx.accounts.auction_house_fee_account.to_account_info(),
+        seller_trade_state: ctx.accounts.seller_trade_state.to_account_info(),
+        free_seller_trade_state: ctx.accounts.free_seller_trade_state.to_account_info(),
+        authority: ctx.accounts.authority.to_account_info(),
+        auctioneer_authority: ctx.accounts.reward_center.to_account_info(),
+        ah_auctioneer_pda: ctx.accounts.ah_auctioneer_pda.to_account_info(),
+        token_program: ctx.accounts.token_program.to_account_info(),
+        system_program: ctx.accounts.system_program.to_account_info(),
+        program_as_signer: ctx.accounts.program_as_signer.to_account_info(),
+        rent: ctx.accounts.rent.to_account_info(),
+    };
+
+    let auctioneer_sell_signer_seeds: &[&[&[u8]]] = &[&[
+        REWARD_CENTER.as_bytes(),
+        auction_house_key.as_ref(),
+        &[reward_center_bump],
+    ]];
+
+    let (sell_ix, sell_account_infos) = make_auctioneer_instruction(AuctioneerInstructionArgs {
+        accounts: auctioneer_sell_ctx_accounts,
+        instruction_data: auctioneer_sell_params.data(),
+        auctioneer_authority: ctx.accounts.reward_center.key(),
+    });
+
+    invoke_signed(&sell_ix, &sell_account_infos, auctioneer_sell_signer_seeds)?;
+    // Perform execute_sale_cpi
+    let mut execute_sale_args = ExecuteSale {
+        execute_sale_params,
+    }
+    .try_to_vec()?;
+    let mut execute_sale_bytes: Vec<u8> = EXECUTE_SALE_SIGHASH.try_to_vec()?;
+    execute_sale_bytes.append(&mut execute_sale_args);
+    let execute_sale_ix = Instruction::new_with_bytes(
+        crate::ID,
+        &execute_sale_bytes,
+        get_execute_sale_account_metas(&ctx),
+    );
+    invoke(
+        &execute_sale_ix,
+        &[
```

Comment:
Can we not use the CPI helpers anchor generates for the program to do in program CPI?

Also consider looking at some of the logic helper metaplex uses for auction house to bundle things into functions without having to invoke CPI. 

We can actually remove the execute_sale once accept_offer and buy_listing are in.